### ISSUE TITLE
Source code handling as note

### ISSUE THREAD

**Issue by Silberling** [twostairs/paperwork#109]
Hi,

I use digital notebooks mostly for snippets / scripts / stuff I can hardly put into a file on my HDD. I need some kind of notebook which is capable of handling (displaying) source code correctly.

At the moment paperwork uses a non-fixed-width font. I'd like to see a solution so I can read source code in my browser.

Maybe using ACE as plugin or some other syntax highlighting would be cute.

**Comment by Baldrs** [comment]
Fortunately, threre are plugins for that
http://ckeditor.com/addon/codesnippet (pure highlight.js)
Using GeSHi http://ckeditor.com/addon/codesnippetgeshi (server-side highlighter)

**Comment by devilx** [comment]
:+1: so, we could use "codesnippet" for the CKEditor and pure highlight.js for displaying the note.

**Comment by sunyoureyes** [comment]
It will be a very cool featur. For example Evernote not have this one.

**Comment by Silberling** [comment]
Thats the main reason why I don't use anything like Evernote: It destroys your code ^^ Gonna try this the next days ;-) @Baldrs

**Comment by Baldrs** [comment]
That's seems to be already done in e393d61

![scrot](https://cloud.githubusercontent.com/assets/814651/5914465/892d8a98-a5fe-11e4-86cc-adc2b50bfd08.png)

**Comment by Baldrs** [comment]
And the third is that note cannot be edited at all

**Comment by devilx** [comment]
Couldn't remember including that plugin when updating CKEditor - but as it seems, I did. Yet, highligh.js isn't included site-wide, therefor highlighting in view-mode doesn't work so far.

I'm wondering why you can't edit the note afterwards. Could you check your web-console and give more insights on this?

**Comment by Baldrs** [comment]
I will file an issue in the evening, I currently don't have paperwork installed on my work machine and my DO droplet ran out of money. It was something in paperwork.min.js

**Comment by devilx** [comment]
@Baldrs remember: paperwork.min.js is being built out of all the sources within js/paperwork/ using gulp.

**Comment by Silberling** [comment]
I just installed paperwork and tried your suggestions.
It works pretty well. Yeah, Highlighting does not work, that's right. But there is a way to insert code blocks and define a language. At the moment I use a mediawiki for this case with a plugin which is pretty nasty.

I am pretty surprised how well paperwork meets the requirements I a missing in so many good applications. But there is still work to do.

For example: Creating a note requires using the menu on the top left. I'd like to have a toolbar instead.

As I said I use mediawiki at the moment. I would have to transfer everything over to paperwork. I would like to have an import option but I think mediawiki is not a priority.

I do like The tagging + Title + Notebook organization. Mediawiki uses Categories. OneNote uses Books, Sections and pages. But There are often cases in which a note fits into multiple of these. That's where I always wished to have tags instead. Pretty cool :-)

I am missing a function to disable registration so this is a private application only.

Having this said I am pretty interested in using Paperwork and trying it in my personal production environment for some time.

As far as the support is done now I would consider this issue as closed. Except for the Highlighting thingy.

**Comment by FromBabylon** [comment]
How to add ckeditor plugins ? 
I've noticed there are some highlighting ones and I was really interested in that feature !

**Comment by Silberling** [comment]
Is there a plugin functionality already? Or did I miss it? Or do I have to install plugins manually?

**Comment by Baldrs** [comment]
@devilx  I rebuilt it and it works now, seems like gulp rebuild is important part of the install process. 

I think now that displayed html should be decoupled from source, because having highlight to highlight the notes dynamically is quite inefficient, and the rendered snippet should be rendered only once.

**Comment by Liongold** [comment]
@Silberling Could you open an issue for your suggestions (one for each suggestion please)?

**Comment by Baldrs** [comment]
Let's plug the highlight js into note display first

**Comment by Baldrs** [comment]
Fixed in #482
